// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package sgdsvm;

// Interface exported by the server.
service SGDSVM{
  // A simple RPC.
  // Obtains the feature at a given position.
  // A feature with an empty name is returned if there's no feature at the given
  // position.
  rpc getDataLabels(Auth) returns (Data) {}
  rpc auth(Empty) returns (Auth) {}
  rpc sendGradientUpdate(GradientUpdate) returns (Empty) {}
  rpc sendDoneComputing(Empty) returns (Empty) {}
}

// Gradient computed
message GradientUpdate {
  string gradient_update_json = 1;
  int32 id = 2;
}

message Data {
  string samples_json = 1;
  string labels_json = 2;
  string weights_json = 3;
}

message Auth {
  int32 id = 1;
}

message Empty {}
